components/script/dom/bluetooth.rs

```diff
@@ -103,36 +215,38 @@ fn canonicalize_filter(filter: &BluetoothScanFilter, global: GlobalRef) -> Falli
         if filter_name_prefix.len() > MAX_FILTER_NAME_LENGTH {
             return Err(Type(FILTER_NAME_TOO_LONG_ERROR.to_owned()));
         }
+
+        // Step 2.4.5.2.
         name_prefix = filter_name_prefix.to_string();
     }
 
-    Ok(BluetoothScanfilter::new(name, name_prefix, services_vec))
-}
+    let mut manufacturer_id = None;
 
-fn convert_request_device_options(options: &RequestDeviceOptions,
-                                  global: GlobalRef)
-                                  -> Fallible<RequestDeviceoptions> {
-    if options.filters.is_empty() {
-        return Err(Type(FILTER_EMPTY_ERROR.to_owned()));
+    // Step 2.4.6.
+    if let Some(manufact_id) = filter.manufacturerId {
+        manufacturer_id = Some(manufact_id);
     }
 
-    let mut filters = vec!();
-    for filter in &options.filters {
-        filters.push(try!(canonicalize_filter(&filter, global)));
-    }
+    let mut service_data_uuid = String::new();
+    // Step 2.4.7.
```

Comment:
I'd prefer the `// Step` comments above the related variable, eg. above `service_data_uuid`.